### PR TITLE
feat(gridlayout) : adding grid layout and tile implementation

### PR DESCRIPTION
**Pull request by jlame**
**What is the problem this PR is trying to solve?**
a draggable grid is needed for TDS monitoring feature.
https://company-57688.frontify.com/document/92132#/navigation-layout/dashboard-monitoring

**What is the chosen solution to this problem?**
using react-grid-layout

**Please check if the PR fulfills these requirements**

* [ ] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/components/src/GridLayout/Grid.component.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+const columns = {
+	xs: 2,
+	s: 4,
+	m: 6,
+	l: 12,
+};
+
+const breakpoints = {
+	xs: 479,
```
Comment (jare-talend) [inline packages/components/src/GridLayout/Grid.component.js]: why precisely 479 ?

File: packages/components/src/GridLayout/Grid.component.js [hunk 2]
```diff
@@ -0,0 +1,119 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import 'react-grid-layout/css/styles.css';
+import { Responsive, WidthProvider } from 'react-grid-layout';
+import Tile from './Tile';
+import { SKELETON_TILE_CONF } from './Tile/Skeleton/SkeletonTile.component';
+// eslint-disable-next-line no-unused-vars
```
Comment (jare-talend) [inline packages/components/src/GridLayout/Grid.component.js]: unecessary ?

File: packages/components/src/GridLayout/Tile/context.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import React, { useContext } from 'react';
+
+// eslint-disable-next-line import/prefer-default-export
+export const TileContext = React.createContext();
+
+export function useTileContext() {
+	const context = useContext(TileContext);
+	if (!context) {
+		throw Error(
```
Comment (jare-talend) [inline packages/components/src/GridLayout/Tile/context.js]: you can use invariant instead to use throw Error.

File: packages/components/src/GridLayout/Grid.component.js (hunk 1 above)
Comment (jlame) [reply packages/components/src/GridLayout/Grid.component.js]: guideline says so

Comment (jsomsanith) [reply packages/components/src/GridLayout/Grid.component.js]: In fact the guideline is not really correct, the small breakpoint should be 0 to 768

File: packages/components/src/GridLayout/Grid.component.js [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+	};
+}
+
+function Grid({
+	children,
+	isResizable = false,
+	layoutChangeCallback,
+	onBreakpointChangeCallback,
+	onDragStopCallback,
+	onResizeStopCallback,
```
Comment (jsomsanith) [inline packages/components/src/GridLayout/Grid.component.js]: Can you change the names to 
* onLayoutChange
* onBreakpointChange
* onDragStop
* onResizeStop

File: packages/components/src/GridLayout/Grid.component.js [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+	isLoading = false,
+	skeletonConfiguration,
+}) {
+	return (
+		<ResponsiveGridLayout
+			className="layout"
+			onLayoutChange={onLayoutChange(layoutChangeCallback)}
+			onDragStop={onDragStop(onDragStopCallback)}
+			onResizeStop={onResizeStop(onResizeStopCallback)}
+			onBreakpointChange={onBreakpointChange(onBreakpointChangeCallback)}
```
Comment (jsomsanith) [inline packages/components/src/GridLayout/Grid.component.js]: You don't need those higher order functions
```javascript
function noOp = () => {};

function Grid({
	children,
	isResizable = false,
	onLayoutChange = noOp,
	onBreakpointChange = noOp,
	onDragStop = noOp,
	onResizeStop = noOp,
	isLoading = false,
	skeletonConfiguration,
}) {
	return (
		<ResponsiveGridLayout
			className="layout"
			onLayoutChange={onLayoutChange}
			onDragStop={onDragStop}
			onResizeStop={onResizeStop}
			onBreakpointChange={onBreakpointChange}
```

File: packages/components/src/GridLayout/Tile/Body/TileBody.component.test.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import Tile from '../index';
+
+describe("tile's body", () => {
+	it('should compute tile body', () => {
+		// given
```
Comment (jsomsanith) [inline packages/components/src/GridLayout/Tile/Body/TileBody.component.test.js]: Please remove that

File: packages/components/src/GridLayout/Tile/Body/TileBody.component.test.js [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+		// when
+		const wrapper = shallow(<Tile.Body>my body</Tile.Body>);
+
+		// then
+		expect(wrapper.getElement()).toMatchSnapshot();
+		// expect(wrapper.getElement()).toHaveLength(1);
```
Comment (jsomsanith) [inline packages/components/src/GridLayout/Tile/Body/TileBody.component.test.js]: Please remove that

File: packages/components/stories/GridLayout.js [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+			key: 'sixthTile',
+			'data-grid': {w: 4, h: 2, x: 4, y: 2, i: 'sixthTile'},
+		},
+	]);
+
+	function layoutChangeCallback(newlayouts) {}
+
+	function onBreakpointChangeCallback(breakpoint, cols) {}
+
+	function onDragStopCallback(layout, oldItem, newItem) {}
```
Comment (jsomsanith) [inline packages/components/stories/GridLayout.js]: Those callbacks are not used. You have 2 choices :
- use storybook actions to log those calls
- don't pass anything and use the default values (see my previous comments)

File: packages/components/src/GridLayout/Grid.component.js (hunk 1 above)
Comment (jlame) [reply packages/components/src/GridLayout/Grid.component.js]: will check with anne

File: packages/components/src/GridLayout/Grid.component.js (hunk 5 above)
Comment (jlame) [reply packages/components/src/GridLayout/Grid.component.js]: yup did it at first cause i thought i might add some behaviour inside UI then call my host app function. but didn't have to in the end.

**Pull request merged**